pkgs/by-name/go/golem/package.nix

```diff
@@ -0,0 +1,62 @@
+{
+  lib,
+  fetchFromGitHub,
+  rustPlatform,
+  pkg-config,
+  openssl,
+  stdenv,
+  darwin,
+  protobuf,
+  redis,
+  fontconfig,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "golem";
+  version = "1.0.16";
+
+  src = fetchFromGitHub {
+    owner = "golemcloud";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-XfSy6MGkEuEAhrF9LQJvHM786fFrBhYNJ4cIX+1eCvo=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    protobuf
+  ];
+  buildInputs =
+    [
+      fontconfig
+      openssl.dev
+    ]
+    ++ lib.optionals stdenv.isDarwin [
+      #darwin.apple_sdk.frameworks.SystemConfiguration
+    ];
+
+  # Tests are failing in the sandbox because of some redis integration tests
+  doCheck = false;
+  checkInputs = [ redis ];
+
+  # Required for golem-wasm-rpc's build.rs to find the required protobuf files
+  # https://github.com/golemcloud/wasm-rpc/blob/v1.0.3/wasm-rpc/build.rs#L7
+  GOLEM_WASM_AST_ROOT = "/build/cargo-vendor-dir/golem-wasm-ast-1.0.0";
```

Comment:
im not the biggest fan of using `/build/cargo-vendor-dir` directly, as its technically an internal implementation detail of the cargo lock stuff, but there doesn't seem to be any other reasonable way to target that directory that wouldn't also IFD.

Reply:
If there is a nicer way to do this I would be happy if someone could point me to it. Making these env-vars available for configuration was required because `cargo_metadata` would try to connect to the internet to update even though a lockfile was present and `--frozen/--offline` was passed to it.

Reply:
yeah i see your use-case for it. its not a blocker, and ultimately making it possible to access through an attribute is out of scope for this PR since it would probably require a bit of refactoring of the rustPlatforms stuff.